app/controllers/all-service-transactions/get-controller.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+// NPM dependencies
+const _ = require('lodash')
+
+const { response } = require('../../utils/response')
+const { ConnectorClient } = require('../../services/clients/connector_client.js')
+const serviceService = require('../../services/service_service')
+const transactionService = require('../../services/transaction_service')
+const { buildPaymentList } = require('../../utils/transaction_view.js')
+const { getFilters, describeFilters } = require('../../utils/filters.js')
+const router = require('../../routes.js')
+const states = require('../../utils/states')
+const client = new ConnectorClient(process.env.CONNECTOR_URL)
+
+const { CORRELATION_HEADER } = require('../../utils/correlation_header.js')
+
+module.exports = async (req, res) => {
+  const correlationId = req.headers[CORRELATION_HEADER] || ''
+  const servicesRoles = _.get(req, 'user.serviceRoles', [])
+  const filters = getFilters(req)
+  try {
+    const aggregatedGatewayAccountIds = servicesRoles
+      .flatMap(servicesRole => servicesRole.service.gatewayAccountIds)
+      .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+    const aggregatedGatewayAccounts = await serviceService.getGatewayAccounts(aggregatedGatewayAccountIds, req.correlationId)
+    const csList = _.flatMap(aggregatedGatewayAccounts, x => x.id).join(',')
+    const searchResultOutput = await transactionService.search(csList, filters.result)
+    client.getAllCardTypes({ correlationId }, allCards => {
```

Comment:
can we use `getAllCardTypesPromise`?  does it return the same?

Reply:
done